src/main/java/uk/gov/companieshouse/authcodenotification/email/KafkaEmailClient.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.companieshouse.authcodenotification.email;
+
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.authcodenotification.exception.ServiceException;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+
+import java.io.IOException;
+import java.time.ZoneId;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+@Component
+public class KafkaEmailClient {
+
+    @Value("${EMAIL_SEND_QUEUE_TOPIC}")
+    private String emailSendQueueTopic;
+
+    private final CHKafkaProducer producer;
+    private final AvroSerializer avroSerializer;
+    private final Schema schema;
+
+    @Autowired
+    public KafkaEmailClient(CHKafkaProducer producer,
+                            AvroSerializer avroSerializer,
+                            Schema schema) {
+        this.producer = producer;
+        this.avroSerializer = avroSerializer;
+        this.schema = schema;
+    }
+
+    public void sendEmailToKafka(EmailContent emailContent)
+            throws ServiceException {
+        try {
+            Message message = new Message();
+            byte[] serializedData = avroSerializer.serialize(emailContent, schema);
+            message.setValue(serializedData);
+            message.setTopic(emailSendQueueTopic);
+            message.setTimestamp(emailContent.getCreatedAt().atZone(ZoneId.systemDefault()).toEpochSecond());
+            Future<RecordMetadata> future = producer.sendAndReturnFuture(message);
+            future.get();
+        } catch (IOException | ExecutionException e) {
+            throw new ServiceException(e.getMessage());
```

Comment:
Personally, I would log the exception at this point - no guarantees that the calling client won't just ignore it, and the error is then hidden.

At the very least the IO/Execution exception that caused this should be supplied as a second parameter to the ServiceException.

Reply:
added extra logging around this